File: src/main/java/club/xiaojiawei/enums/RunModeEnum.java
package club.xiaojiawei.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;


/**
 * @author 肖嘉威
 * @date 2023/7/6 20:20
 */
@Getter
@ToString
@AllArgsConstructor
public enum RunModeEnum {

    STANDARD("标准模式", ModeEnum.TOURNAMENT, true),
    WILD("狂野模式", ModeEnum.TOURNAMENT, true),
    CASUAL("休闲模式", ModeEnum.TOURNAMENT, true),
    CLASSIC("经典模式", ModeEnum.TOURNAMENT, false),
    TWIST("幻变模式", ModeEnum.TOURNAMENT, false),
    BACON("酒馆战棋", ModeEnum.BACON, false),
    ;

    private final String comment;
    private final ModeEnum modeEnum;
    private final boolean enable;

}

File: Hearthstone-Script-Base/src/main/kotlin/club/xiaojiawei/enums/RunModeEnum.kt
package club.xiaojiawei.enums

import club.xiaojiawei.config.log

/**
 * @author 肖嘉威
 * @date 2023/7/6 20:20
 */
enum class RunModeEnum(
    val comment: String,
    val modeEnum: ModeEnum,
    var isEnable: Boolean,
) {
    STANDARD("标准模式", ModeEnum.TOURNAMENT, true),
    WILD("狂野模式", ModeEnum.TOURNAMENT, true),
    CASUAL("休闲模式", ModeEnum.TOURNAMENT, true),
    PRACTICE("练习模式", ModeEnum.ADVENTURE, true),
    CLASSIC("经典模式", ModeEnum.TOURNAMENT, false),
    TWIST("幻变模式", ModeEnum.TOURNAMENT, false),
    BACON("酒馆战棋", ModeEnum.BACON, false),
    ;

    companion object {
        fun fromString(string: String): RunModeEnum? =
            try {
                valueOf(string)
            } catch (_: Exception) {
                log.warn { "未适配$string" }
                null
            }
    }
}
